iceoryx_examples/icediscovery_in_c/iox_c_find_service.c

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_binding_c/runtime.h"
+//! [include service discovery]
+#include "iceoryx_binding_c/service_discovery.h"
+//! [include service discovery]
+#include "sleep_for.h"
+
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+#define SEARCH_RESULT_CAPACITY 10
+
+bool keepRunning = true;
+
+const char APP_NAME[] = "iox-c-find-service";
+
+static void sigHandler(int signalValue)
+{
+    // Ignore unused variable warning
+    (void)signalValue;
+    // caught SIGINT or SIGTERM, now exit gracefully
+    keepRunning = false;
+}
+
+//! [print function to be applied to search results]
+void printSearchResult(const iox_service_description_t service)
+{
+    printf(
+        "- Service: %s, Instance: %s, Event: %s\n", service.serviceString, service.instanceString, service.eventString);
+}
+//! [print function to be applied to search results]
+
+//! [search function for all front devices]
+void searchFrontDevices(const iox_service_description_t service, void* count)
+{
+    if (strncmp(service.instanceString, "FrontLeft", IOX_CONFIG_SERVICE_STRING_SIZE - 1U) == 0
```

Comment:
This `-1U` here on user side is unfortunate and error-prone. Do we want to define the string size *without*  the zero terminator included and compensate with a +1 in the tests instead where needed?

Or even have two constants, i.e. `IOX_CONFIG_SERVICE_STRING_SIZE_WITH_ZERO_TERMINATON` in addition.

From https://www.cplusplus.com/reference/cstring/strncmp/ I am not sure we really need to subtract 1 here though for correctness. It will not exceed the buffer either way. In this case we would only drop the `-1U`.